### DISCUSSION TITLE
How can i receive request (headers, host, absolute url) in my custom class (service)?

### DISCUSSION THREAD

**Discussion by sasha172017** [adonisjs/core#1745] [Help] [answered]
I need get request in custom class, how can i do to? does have adonis global request, i neet it NOT in controller?

**Comment by thetutlage** [comment, marked as answer]
There is no global request in AdonisJS and not possible in Node.js at all. You will have to pass it to the service constructor or the method you are calling

**Reply by sasha172017** [reply]
thank you for answer)
